webapp/src/app.tsx

```diff
@@ -2144,16 +2154,15 @@ export class ProjectView
         pxt.log("turning debugging mode to " + state);
         this.setState({ debugging: state, tracing: false }, () => {
             this.renderCore()
+            const blocks = this.blocksEditor.editor.getAllBlocks();
```

Comment:
it's not guaranteed that the blocks editor is loaded yet. You should not leak this abstraction here.